tests/integrationv2/utils.py

```diff
@@ -51,6 +51,28 @@ def get_parameter_name(item):
     return str(item)
 
 
+def expected_signature(protocol, signature):
```

Comment:
These functions seem specific to what you're testing, so I'm not sure they make sense in utils.py. Especially the TLS 1.2 fallback one should be test_signature_algorithms.py I think.